raven/utilities/forecasting.py

```diff
@@ -173,3 +182,98 @@ def perform_climatology_esp(model_name, forecast_date, forecast_duration, **kwds
     qsims["member"] = (["member"], avail_years)
 
     return qsims
+
+
+def get_recent_ECCC_forecast(region, climate_model='GEPS'):
+    '''
+    This function generates a forecast dataset that can be used to run raven. 
+    Data comes from the ECCC datamart and collected daily. It is aggregated 
+    such that each file contains forecast data for a single day, but for all 
+    forecast timesteps and all members.
+    
+    The code takes the region shapefile and the climate_model to use, here GEPS
+    by default, but eventually could be GEPS, GDPS, REPS or RDPS.
+    '''
+   
+    # Get the file locations and filenames as a function of the climate model and date
+    if climate_model is 'GEPS':
+        
+        # Forecasts will always be recent so start with today!
+        date_now=dt.datetime.now()
+
+        # Find the most recent data on the server by looping from today and go
+        #backwards 5 days. If no data is present, not much value in forecasting.
+        filein=None
+        for i in range(0,7):
+            
+            # Write the string
+            date=date_now-dt.timedelta(days=i)
+            datefcst=dt.datetime.strftime(date,"%Y%m%d")
+            
+            # try to open the files. If it works, great, if no
+            try:
+                filein='https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/dodsC/birdhouse/testdata/geps_forecast/CMC_geps-raw_latlon0p5x0p5_' + datefcst + '00_allP_allmbrs.nc'
+                ds=xr.open_dataset(filein)
+                # Checking that these exist. IF the files are still processing, possible that one or both are not available!
+                assert 'pr' in ds
+                assert 'tas' in ds
+                break
+            except:
+                filein=None
```

Comment:
```suggestion
```suggestion
                if hasattr(ds, "pr") and hasattr(ds, "tas"):
                    break
                raise AttributeError
            except AttributeError:
                filein=None
```

Reply:
Question: are `pr` and `tas` supposed to be present only in the `GEPS` dataset, or in all? (GDPS, RDPS, REPS)

Reply:
they will be in all the climate model files.